lib/trento/activity_logging/parser/metadata_enricher.ex

```diff
@@ -79,7 +69,31 @@ defmodule Trento.ActivityLog.Logger.Parser.MetadataEnricher do
   defp detect_enrichment(:cluster, {_, %{cluster_id: id}}), do: {:ok, id}
   defp detect_enrichment(:database, {_, %{database_id: id}}), do: {:ok, id}
   defp detect_enrichment(:sap_system, {_, %{sap_system_id: id}}), do: {:ok, id}
+  defp detect_enrichment(:user, {_, %{user_id: id}}), do: {:ok, id}
+
+  defp detect_enrichment(
+         target_entity,
+         {activity,
+          %{
+            resource_id: resource_id,
+            resource_type: target_entity
+          }}
+       )
+       when activity in [:resource_tagging, :resource_untagging],
+       do: {:ok, resource_id}
 
   defp detect_enrichment(_target_entity, {_activity, _metadata}),
     do: {:error, :no_enrichment_needed}
+
+  defp polish_entity(%{username: username, deleted_at: deleted_at} = entity) do
+    # If the user is deleted, we append the deletion date to the username
+    # It's a implementation detail that we don't want to expose to the user
+    # See Trento.Users context for more information
+    clean_username =
+      String.trim_trailing(username, "__" <> DateTime.to_string(deleted_at))
+
```

Comment:
I wonder: do we need to guard against the `deleted_at` value being `nil` ?

Reply:
It should never happen but I see no harm in doing it.